src/Microsoft.XmlSerializer.Generator/pkg/build/Microsoft.XmlSerializer.Generator.targets

```diff
@@ -0,0 +1,27 @@
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <SerializerName>$(AssemblyName).XmlSerializers</SerializerName>
+    <SerializerDllName>$(SerializerName).dll</SerializerDllName>
+    <SerializerPdbName>$(SerializerName).pdb</SerializerPdbName>
+    <SerializerCsName>$(SerializerName).cs</SerializerCsName>
+    <SerializerDllImmediatePath>$(IntermediateOutputPath)$(SerializerName).dll</SerializerDllImmediatePath>
+    <SerializerPdbImmediatePath>$(IntermediateOutputPath)$(SerializerName).pdb</SerializerPdbImmediatePath>
+    <SerializerCsImmediatePath>$(IntermediateOutputPath)$(SerializerName).cs</SerializerCsImmediatePath>
+    <SGenWarningText>SGEN : warning SGEN1: Fail to generate the serializer for $(AssemblyName).dll. Please check the build log for the details.</SGenWarningText>
+  </PropertyGroup>
+  <Target Name="GenerateSerializationAssembly" AfterTargets="Build">
+    <Delete Condition="Exists('$(SerializerCsImmediatePath)') == 'true'" Files="$(SerializerDllImmediatePath)" />
```

Comment:
Copy-paste error. The condition should check `SerializerDllImmediatePath` instead of `SerializerCsImmediatePath` .